### PR TITLE
Move rdoc and psych into their own bundler group

### PR DESCRIPTION
**Pull request by stevecheckoway**
Psych 5 no longer bundles libyaml which causes `bundle install` to fail on macOS which doesn't have libyaml and doesn't have an easy, standard way to install it.

One can use `bundle config set --local without rdoc` to disable installing the `psych` and `rdoc` (which depends on `psych`) gems.

Hopefully, this is only temporary and the upstream Psych team will produce a better way to handle this.

**What problem is this PR intended to solve?**

#2769

**Have you included adequate test coverage?**

As this is a (hopefully temporary) workaround, rather than a supported configuration, I have not.

**Does this change affect the behavior of either the C or the Java implementations?**

No.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -36,3 +34,11 @@ group :development do
     gem "rubocop-shopify", "2.10.1"
   end
 end
+
+# If Psych doesn't build, you can disable this group locally by running
+# `bundle config set --local without rdoc`
+# Then re-run `bundle install`.
+group :rdoc do
+  gem "rdoc", "6.5.0"
+  gem "psych", "~> 5.0" # psych 5 isn't building in places yet https://github.com/ruby/setup-ruby/issues/409
```
Comment (flavorjones) [inline Gemfile:43]: I think we should remove the `psych` line now. It was originally added in 6a3254cd to pin to `~> 4.0` but that was later updated by dependabot to `~> 5.0` in c7a78029. We actually don't care about Psych as long as it stays out of our way!

**Review by flavorjones — CHANGES_REQUESTED** [review]

**Comment by stevecheckoway** [comment]
I don't recall if you want me to squash and force push to the branch or if you use the "squash and merge" button normally.

**Comment by flavorjones** [comment]
I generally use the "Merge pull request" to create a merge commit, but I don't feel very strongly about it. I'll hit that button now!

**Comment by flavorjones** [comment]
Oh, but if you were asking if I usually force-push to the branch to have a clean history before merging, the answer is "yes"!

**Comment by stevecheckoway** [comment]
Great. I'll do that in the future.

**Pull request merged**